### ISSUE TITLE
[DataGrid] Make actions menu accessible

### ISSUE THREAD

**Issue by m4theushw** [mui/mui-x#3948]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Summary 💡

The menu added by the `actions` column type is not accessible. Once open, keyboard arrows don't work. Pressing <kbd>Esc</kbd> should close it.

<img width="300" src="https://user-images.githubusercontent.com/42154031/153870482-1c5dc7ea-3729-42b3-9f91-21a45b539092.png" alt="image" style="max-width: 100%;">

Related to https://github.com/mui/mui-x/pull/3375#issuecomment-1038820718

**Comment by m4theushw** [comment]
Blocked by https://github.com/w3c/aria-practices/issues/155#issuecomment-1062165181

Until https://www.w3.org/TR/wai-aria-practices/#gridNav_inside docs doesn't specify how to handle cells containing widgets relaying on <kbd>Enter</kbd> I don't think we can offer a great experience. Related to #1465

**Comment by Pustelto** [comment]
Hi, is it necessary to actually wait for Authoring practices to specify this? From my understanding that document is not an official spec, more a guideline. And it is really a question of when that issue will be resolved given it is open from 2016. Maybe MUI could introduce some pattern for the time being to solve this and when ARIA authoring practices are updated change it?

**Comment by m4theushw** [comment]
@Pustelto I opened #4325 with a proposal wrapping the buttons in a div with `role="menu"`. NVDA reads correctly the number of items now. Do you want to give your feedback? https://deploy-preview-4325--material-ui-x.netlify.app/components/data-grid/columns/#column-types To navigate between the items use only the arrow keys.

**Comment by Pustelto** [comment]
Thanks, will have a look as soon as possible.

**Comment by Pustelto** [comment]
@m4theushw I did a quick check, sorry for taking so long. Keyboard navigation is definitely much better. However, when I test it with macOS VoiceOver I have a hard time opening the menu using Cmd+Space on the header. Sometimes doesn't even work, when it does I can't navigate the menu at all using keyboard. if NVDA worked, then it seems like VoiceOver handles the focus slightly differently.

**Comment by m4theushw** [comment]
> I did a quick check, sorry for taking so long. Keyboard navigation is definitely much better. However, when I test it with macOS VoiceOver I have a hard time opening the menu using Cmd+Space on the header. 

@Pustelto Are you talking about the column menu? If yes, then I would ask to open another issue with the problems you found. I do think there's room improvement here too. #3948 is exclusively for the accessibility of the menu from the `actions` column type.

**Comment by Pustelto** [comment]
I see, my bad. Will open another issue then. Thanks for clarification.
